common/base_cpp/array.h

```diff
@@ -37,19 +37,15 @@ namespace indigo
     public:
         DECL_TPL_ERROR(ArrayError);
 
-        explicit Array()
+        Array() : _reserved(0), _length(0), _array(NULL)
         {
-            _reserved = 0;
-            _length = 0;
-            _array = NULL;
         }
 
         ~Array()
         {
             if (_array != NULL)
             {
                 free(_array);
-                _array = NULL;
```

Comment:
Don't see the reason to delete this line. It may potentially lead to double-free and memory corruption.

Reply:
Never after the destructor. :)